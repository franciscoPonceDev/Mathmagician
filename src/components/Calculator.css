#calculator {
  margin: 1rem;
  width: 33%;
  display: grid;
  gap: 1px 1px;
  background-color: #c8c8c8;
  grid-template: repeat(6, 1fr) / repeat(4, 1fr);
}

.display {
  display: flex;
  grid-area: 1 / 1 / span 1 / span 4;
  background-color: rgb(133, 134, 147);
  padding: 1rem;
  font-size: 2rem;
  justify-content: flex-end;
  color: #fff;
}

button {
  border: 1px solid #c8c8c8;
  font-size: 1.5rem;
}

.btn-orange {
  background-color: rgb(232, 143, 62);
}

.btn-gray {
  background-color: rgb(244, 244, 244);
}

.zero {
  grid-area: 6 / 1 / span 1 / span 2;
}
