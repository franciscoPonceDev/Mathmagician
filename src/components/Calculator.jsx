import React from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

export default class Calculator extends React.Component {
  constructor() {
    super();
    this.clickHandler = this.clickHandler.bind(this);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  clickHandler(event) {
    this.setState((state) => calculate(state, event.target.textContent));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <div id="calculator" className="calculator">
        <div className="display">
          {total || '0'}
          {operation || ''}
          {next || ''}
        </div>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          AC
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          +/-
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          %
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-orange">
          รท
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          7
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          8
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          9
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-orange">
          x
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          4
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          5
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          6
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-orange">
          -
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          1
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          2
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          3
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-orange">
          +
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray zero">
          0
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-gray">
          .
        </button>
        <button onClick={this.clickHandler} type="button" className="btn-orange">
          =
        </button>
      </div>
    );
  }
}
