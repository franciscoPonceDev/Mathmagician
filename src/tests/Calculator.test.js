import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('calculator renders correct', () => {
  const tree = renderer.create(<Calculator />).toJSON;

  expect(tree).toMatchSnapshot();
});
