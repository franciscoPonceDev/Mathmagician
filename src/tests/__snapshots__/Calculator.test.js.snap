// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator renders correct 1`] = `[Function]`;

exports[`calculator renders correctly 1`] = `[Function]`;
